src/lib/server/subscriptionUtils.ts

```diff
@@ -27,9 +27,24 @@ export const getStripeCustomerWithSubscriptions = async (uid: string) => {
         expand: ['subscriptions']
     });
 }
+
+export const getAllCustomerSubscriptions = async (uid: string) => {
+    const stripeCustomerId: string | null = (await db.ref(`/users/${uid}/private/stripeCustomerId`).once('value')).val();
+    if (stripeCustomerId === null)
+        return stripeCustomerId;
+    return await stripe.subscriptions.list({
```

Comment:
Will this throw an exception if the customer doesn't exist? That would put pressure on our DB to stay valid

Reply:
The shell version at least does. This is something to think about because it makes the connection between our DB and Stripe kinda brittle. We have three options I think:
1. Accept that it is kinda brittle and be careful whenever we delete Stripe customers to also update Firebase and keep them in sync
2. Catch the error (if it is indeed thrown), check that it is a "not found" error, then safely return an empty array. 
3. Go back to calling this synchronously after fetching the customer, that way we don't call it at all if the Stripe customer isn't found